src/Data/ByteString/Lazy/Base16.hs

```diff
@@ -171,5 +171,7 @@ isBase16 bs = isValidBase16 bs && isRight (decodeBase16Untyped bs)
 -- True
 --
 isValidBase16 :: ByteString -> Bool
-isValidBase16 = all (`elem` "0123456789abcdefABCDEF")
+isValidBase16 bs = case bs of
+    Empty -> True
+    Chunk sbs next -> B16.isValidBase16 sbs && isValidBase16 next
```

Comment:
For my own sanity 

```suggestion
    Chunk lbs next -> B16.isValidBase16 lbs && isValidBase16 next
```

Reply:
The chunk itself is a strict bytestring, though, so this confuses me.

Reply:
Oh true i'm being dumb - but still, `sbs` makes me think `ShortByteString`.